src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
```

Comment:
Don't begin interface or method implementation with an empty line. It's better to have the code start immediately after the interface declaration to follow the Java coding conventions.